Doc/library/gzip.rst

```diff
@@ -143,6 +143,10 @@ The module defines the following items:
       :func:`time.time` and the :attr:`~os.stat_result.st_mtime` attribute of
       the object returned by :func:`os.stat`.
 
+   .. attribute:: name
+
+      The filename.
```

Comment:
It would be nice to have a little bit more explicit detail here.  Is it new in Python 3.12, or simply an oversight and thus should be back-ported to previous versions? And this is the filename of...what, exactly? Is `name` a `str`, a `Path` or something else? Is the extension included?

Reply:
This was added to fix the warning here from `3.12.rst`:

```rst
* :mod:`gzip`: Remove the ``filename`` attribute of :class:`gzip.GzipFile`,
  deprecated since Python 2.6, use the :attr:`~gzip.GzipFile.name` attribute
  instead. In write mode, the ``filename`` attribute added ``'.gz'`` file
  extension if it was not present.
  (Contributed by Victor Stinner in :gh:`94196`.)
```

---

The input argument is documented:

> The filename argument can be an actual filename (a [str](https://docs.python.org/3/library/stdtypes.html#str) or [bytes](https://docs.python.org/3/library/stdtypes.html#bytes) object), or an existing file object to read from or write to.

https://docs.python.org/3/library/gzip.html#gzip.open

---

It's set here:

https://github.com/python/cpython/blob/ca066bdbed85094a9c4d9930823ce3587807db48/Lib/gzip.py#L191

But after these conversions:

https://github.com/python/cpython/blob/ca066bdbed85094a9c4d9930823ce3587807db48/Lib/gzip.py#L175-L182

---


Looks like it's been around since 2007, 2.7 + 3.0: https://github.com/python/cpython/commit/cf297e46b85257396560774e5492e9d71a40f32e

---

So maybe something like this?

```suggestion
      The file system representation of the path, either `str` or `bytes`.
```

Reply:
The current description doesn't make clear if it's just the filename (as the name would imply), the full absolute path, the normalized relative path (one of which the above would imply) or just the path exactly as originally entered, when it is in fact the very latter. Furthermore, at least on Windows it doesn't normalize the directory separators, either:

````python
>>> gzip.GzipFile(filename=Path("../proposals/test_file.gz"), mode="w").name
'../proposals/test_file.gz'
>>> gzip.GzipFile(filename=Path("../proposals/test_file.gz"), mode="w").name
'..\\proposals\\test_file.gz'
>>> gzip.GzipFile(filename=r"..\proposals\test_file.gz", mode="w").name
'..\\proposals\\test_file.gz'
```

So it's important that code that works with it normalize it themselves (e.g. using `Path(gz_file.name).resolve().expanduser()`) to ensure consistent results. , Also, "the file system representation of paths" would likely imply this sort of normalization, when this is actually about converting path-like objects to a str/bytes that can be passed to OS APIs.

Therefore, I'd suggest something like

```suggestion
      The path to the gzip file on disk, as a :class:`str` or :class:`bytes`.
      Equivalent to the output of :func:`os.fspath` on the original input path,
      with no other normalization, resolution or expansion.
```

Reply:
It should, really, be backported, but unfortunately that won't be possible unless it's split off from this PR.

Reply:
Will add a TODO to create a backport for 3.11/3.10 after merge.